tools/apps/src/lib.rs

```diff
@@ -955,6 +955,10 @@ impl HttpConn for Http3Conn {
                     {
                         Ok(v) => v,
 
+                        Err(quiche::h3::Error::Done) => {
```

Comment:
Thinking more about this, I think this actually needs to "return" the value 0, instead of skipping the loop, otherwise the partial response object is not created below.
```suggestion
                        Err(quiche::h3::Error::Done) => 0,
```